src/httpInterceptor.js

```diff
@@ -7,12 +7,15 @@ function httpEtagInterceptorFactory () {
     if (itemCache) {
       var cacheInfo = itemCache.info()
       var cacheResponseData = cacheInfo.cacheResponseData
+      var cacheResponseHeaders = cacheInfo.cacheResponseHeaders
       var etag = response.headers().etag
       var cacheData = {}
 
       if (etag) {
         cacheData.etagHeader = etag
         if (cacheResponseData) cacheData.responseData = response.data
+        // TODO: Allow user to speficy which headers to cache?
+        if (cacheResponseHeaders) cacheData.responseHeaders = response.headers
```

Comment:
`cacheData.responseHeaders` can't be a function. In the case of local and session storage caches, `JSON.stringify` is used, and functions can't be stringified.